src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDKTest.java

```diff
@@ -0,0 +1,218 @@
+
+        Indexable securityGroupIndexable = Mockito.mock(Indexable.class);
+        Observable<Indexable> observableSecurityGroupSuccess = AzureTestUtils.createSimpleObservableSuccess(securityGroupIndexable);
+        Mockito.doReturn(observableSecurityGroupSuccess).when(this.azureVirtualNetworkOperationSDK)
+                .buildCreateSecurityGroupObservable(Mockito.eq(azureCreateVirtualNetworkRef), Mockito.eq(this.azure));
+
+        Mockito.doThrow(new RuntimeException()).when(this.azureVirtualNetworkOperationSDK)
+                .doNetworkCreationStepTwoSync(Mockito.eq(securityGroupIndexable),
+                        Mockito.eq(azureCreateVirtualNetworkRef), Mockito.eq(this.azure));
+
+        // exercise
+        Observable<Indexable> virtualNetworkCreationObservable = this.azureVirtualNetworkOperationSDK
+                .buildVirtualNetworkCreationObservable(azureCreateVirtualNetworkRef, this.azure);
+        virtualNetworkCreationObservable.subscribe();
+
+        // verify
+        this.loggerAssert
+                .assertEqualsInOrder(Level.INFO, Messages.Info.FIRST_STEP_CREATE_VNET_ASYNC_BEHAVIOUR)
+                .assertEqualsInOrder(Level.ERROR, Messages.Error.ERROR_CREATE_VNET_ASYNC_BEHAVIOUR)
+                .assertEqualsInOrder(Level.INFO, Messages.Info.END_CREATE_VNET_ASYNC_BEHAVIOUR);
+    }
+
+    // test case: When calling the buildVirtualNetworkCreationObservable method and the observables execute
+    // without any error, it must verify if It returns the right observable.
+    @Test
+    public void testBuildCreateSecurityGroupObservableSuccessfully() {
+        // set up
+        String nameExpected = TestUtils.ANY_VALUE;
+        String cidrExpected = TestUtils.ANY_VALUE;
+        String resourceGroupNameExpected = TestUtils.ANY_VALUE;
+        AzureCreateVirtualNetworkRef azureCreateVirtualNetworkRef = AzureCreateVirtualNetworkRef.builder()
+                .name(nameExpected)
+                .resourceGroupName(resourceGroupNameExpected)
+                .cidr(cidrExpected)
+                .build();
+        Region regionExpected = Region.fromName(this.regionName);
+
+        Observable<Indexable> observableExpected = Mockito.mock(Observable.class);
+        PowerMockito.mockStatic(AzureNetworkSDK.class);
+        PowerMockito.when(AzureNetworkSDK.createSecurityGroupAsync(Mockito.eq(this.azure), Mockito.eq(nameExpected)
+                , Mockito.eq(regionExpected), Mockito.eq(resourceGroupNameExpected), Mockito.eq(cidrExpected)))
+                .thenReturn(observableExpected);
+
+        // exercise
+        Observable<Indexable> observable = this.azureVirtualNetworkOperationSDK.buildCreateSecurityGroupObservable(azureCreateVirtualNetworkRef, this.azure);
+
+        // verify
+        Assert.assertEquals(observableExpected, observable);
+    }
+
+    // test case: When calling the buildVirtualNetworkCreationObservable method and the observables execute
```

Comment:
The method name present in the test case description doesn't match with the method tested.

Reply:
Done.
Note: I have fixed it in the last branch of Azure Network context.